lib/lightning/projects/project.ex

```diff
@@ -40,7 +40,7 @@ defmodule Lightning.Projects.Project do
     has_many :project_credentials, ProjectCredential
     has_many :credentials, through: [:project_credentials, :credential]
 
-    has_many :workflows, Workflow
+    has_many :workflows, Workflow, where: [deleted_at: nil]
```

Comment:
Whoa. I wasn't expecting this to be the solution. Isn't the trouble also that when individual _jobs_ inside a workflow have been deleted they show up inside the `state.json`?